src/smolagents/memory.py

```diff
@@ -17,6 +19,9 @@ class Message(TypedDict):
     role: MessageRole
     content: str | list[dict]
 
+    def dict(self):
+        return {"role": self.role, "content": self.content}
+
```

Comment:
This method is not necessary: the class instance is already a dict. See: https://github.com/huggingface/smolagents/pull/316#discussion_r1930601555